ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/SessionProperties.java

```diff
@@ -89,7 +92,11 @@ public KsqlHostInfo getKsqlHostInfo() {
   }
 
   public URL getLocalUrl() {
-    return localUrl;
+    try {
+      return new URL(localUrl.toString());
+    } catch (final MalformedURLException fatalError) {
+      throw new IllegalStateException("Could not deep copy URL: " + localUrl);
+    }
```

Comment:
This is a nasty one. Not straightforward to make a deep-copy (cf https://github.com/confluentinc/ksql/pull/7812/files#r671882647)